cpp/tests/community/k_truss_test.cu

```diff
@@ -0,0 +1,408 @@
+                thrust::host,
+                weighted_edge_first,
+                weighted_edge_first + h_src.size(),
+                [edge](auto e) {
+                  auto src         = thrust::get<0>(e);
+                  auto dst         = thrust::get<1>(e);
+                  auto remove_edge = thrust::make_tuple(src, dst);
+                  return (remove_edge == edge || thrust::make_tuple(dst, src) == edge);
+                }
+
+              );
+
+              h_src.resize(thrust::distance(weighted_edge_first, weighted_edge_last));
+              h_dst.resize(thrust::distance(weighted_edge_first, weighted_edge_last));
+              h_wgt.resize(thrust::distance(weighted_edge_first, weighted_edge_last));
+            }
+            auto edge_last = thrust::remove_if(
+              thrust::host,
+              edge_first,
+              edge_first + h_src.size(),
+              [edge](auto e) {
+                auto src         = thrust::get<0>(e);
+                auto dst         = thrust::get<1>(e);
+                auto remove_edge = thrust::make_tuple(src, dst);
+                return (remove_edge == edge || thrust::make_tuple(dst, src) == edge);
+              }
+
+            );
+
+            h_src.resize(thrust::distance(edge_first, edge_last));
+            h_dst.resize(thrust::distance(edge_first, edge_last));
+
+            n_dropped += 1;
+          }
+        }
+      }
+    }
+
+    // convert back to device_vector
+    rmm::device_uvector<vertex_t> d_reference_src(h_src.size(), handle.get_stream());
+
+    raft::update_device(d_reference_src.data(), h_src.data(), h_src.size(), handle.get_stream());
+
+    rmm::device_uvector<vertex_t> d_reference_dst(h_src.size(), handle.get_stream());
+
+    raft::update_device(d_reference_dst.data(), h_dst.data(), h_dst.size(), handle.get_stream());
+
+    auto d_reference_wgt =
+      std::make_optional<rmm::device_uvector<weight_t>>(h_src.size(), handle.get_stream());
+
+    if (edge_weight_view) {
+      raft::update_device(
+        (*d_reference_wgt).data(), h_wgt.data(), h_wgt.size(), handle.get_stream());
+    }
+
+    return std::make_tuple(std::move(d_reference_src),
+                           std::move(d_reference_dst),
+                           edge_weight_view ? std::move(d_reference_wgt) : std::nullopt);
+  }
+
+  template <typename vertex_t, typename edge_t, typename weight_t>
+  void run_current_test(std::tuple<KTruss_Usecase const&, input_usecase_t const&> const& param)
+  {
+    constexpr bool renumber = false;
+
+    auto [k_truss_usecase, input_usecase] = param;
+
+    raft::handle_t handle{};
+
+    HighResTimer hr_timer{};
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.start("MG Construct graph");
+    }
+
+    // NX k_truss is not implemented for graph with self loop and isolated
+    // vertices therefore dropped them especially for rmat generated graphs
```

Comment:
Can we just drop self-loops in the reference K-Truss implementation? (std::remove_if)

Reply:
The datasets files I tested do not have self loops and for the rmat generated, I set flags to drops self loops and multi edges.